backend/src/main/java/ca/mcgill/purposeful/service/IdeaService.java

```diff
@@ -472,4 +474,59 @@ public void removeIdeaById(String uuid) {
     // remove idea
     ideaRepository.deleteById(uuid);
   }
+
+  /**
+   * Retrieve all the ideas that a user expressed interest in.
+   *
+   * @param email EMail of the user to retrieve the ideas for
+   * @return List of ideas that the user expressed interest in
+   */
+  @Transactional
+  public List<Idea> getIdeasByCollaborationRequest(String email) {
+    // Input validation
+    String error = "";
+    if (email == null || email.trim().length() == 0) {
+      error += "Email cannot be left empty! ";
+    }
+    if (error.length() > 0) {
+      throw new GlobalException(HttpStatus.BAD_REQUEST, error);
+    }
+
+    // Check if the user we are trying to modify does indeed exist
+    RegularUser user = regularUserRepository.findRegularUserByAppUserEmail(email);
+    if (user == null) {
+      throw new GlobalException(HttpStatus.BAD_REQUEST, "This account does not exist.");
+    }
+
+    // Get all Collaboration Requests ever sent
+    Iterable<CollaborationRequest> allRequests = collaborationRequestRepository.findAll();
+
+    // Check whether the request was successful
+    if (allRequests == null) {
+      throw new GlobalException(HttpStatus.INTERNAL_SERVER_ERROR, "Could not retrieve requests.");
```

Comment:
Could this cause a problem initially when there are no collaboration requests? I think we don't need this check, or if we keep it we change error message to "There are currently no collaboration requests in the database". To check the functionality of the findall() method, we can add it to the repo tests to make sure its working.

EDIT: After going through tests, I think we should remove this check. The repo tests can be used to make sure the findAll() method works, you can add a line there that uses it and asserts its not null and that it contains the collab request. The file is in test/dao called CollaborationConfirmationRepositoryTests